spec/concerns/reset_vacancies_in_job_listings_spec.rb

```diff
@@ -224,4 +224,17 @@
         .from(previous_further_details).to(nil)
     end
   end
+
+  context "when changing benefits" do
+    subject(:update_benefits) { vacancy.update(benefits: false) }
+
+    let(:vacancy) { create(:vacancy, benefits: true, benefits_details: "test") }
```

Comment:
Apparently, unless you need a record to be persisted, `build_stubbed` is the best option to go for. Makes tests quicker! https://thoughtbot.com/blog/use-factory-bots-build-stubbed-for-a-faster-test